src/core/fns.js

```diff
@@ -0,0 +1,102 @@
+var types = require('./types'),
+	type = types.type,
+	FUNCTION = types.FUNCTION,
+	ARRAY = types.ARRAY;
+	OBJECT = types.OBJECT;
+
+var noop = function(){};
+
+function propify(promise, initialValue) {
+		var prop = m.prop(initialValue);
```

Comment:
Nit: you might want to add this to the top: `var m = require('../modules/utils')`. Otherwise, it's broken unless `m` is globally installed.